packages/evm/lib/nft/EvmNftProvider.ts

```diff
@@ -1,32 +1,34 @@
 import { ClientTypes, HttpClient, Nft } from '@chainify/client';
 import { UnsupportedMethodError } from '@chainify/errors';
 import { AddressType, BigNumber, FeeType, Transaction } from '@chainify/types';
+import { AbiCoder } from '@ethersproject/abi';
 import { Signer } from '@ethersproject/abstract-signer';
 import { AddressZero } from '@ethersproject/constants';
 import { BaseProvider } from '@ethersproject/providers';
 import { ERC1155, ERC1155__factory, ERC721, ERC721__factory } from '../typechain';
-import { EthersPopulatedTransaction, EthersTransactionResponse, NftTypes } from '../types';
+import { EthersPopulatedTransaction, EthersTransactionResponse, NFTAsset, NftTypes } from '../types';
 import { toEthereumTxRequest } from '../utils';
 import { EvmBaseWalletProvider } from '../wallet/EvmBaseWalletProvider';
 
 type NftContract = ERC721 | ERC1155;
 type NftInfo = { contract: NftContract; schema: NftTypes };
 
-export class EvmNftProvider extends Nft<BaseProvider, Signer> {
+export abstract class EvmNftProvider extends Nft<BaseProvider, Signer> {
     private _erc721: ERC721;
     private _erc1155: ERC1155;
-    private _cache: Record<string, NftInfo>;
-    private _schemas: Record<string, NftContract>;
-    private _httpClient: HttpClient;
+
+    protected schemas: Record<string, NftContract>;
+    protected cache: Record<string, NftInfo>;
+    protected httpClient: HttpClient;
 
     constructor(walletProvider: EvmBaseWalletProvider<BaseProvider>, httpConfig: ClientTypes.AxiosRequestConfig) {
```

Comment:
replace `httpConfig` with the parameters that client needs to pass. e.g. `url`, `apiKey`